File: 2 CPP/Leetcode-1021-RemoveOutermostParentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string result = "";
        int opened = 0;
        for(int i=0;i<s.size();++i){
            char c = s[i];
            if(c=='('){
                if (opened>0) result+=c;
                opened+=1;
            }
            else{
                opened-=1;
                if (opened>0) result+=c;
            }
        }
        return result;
    }
};

File: 1 Python/Leetcode-1021-RemoveOutermostParentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        openCt = 0
        closeCt = 0
        copied = list(s)        
        for i in range(len(copied)):
            c = copied[i]
            if c=="(":
                openCt+=1
            else:
                closeCt+=1

            if openCt==closeCt:
                copied[i-(openCt*2-1)]=""
                copied[i]=""
                openCt=0
                closeCt=0

        return "".join(copied)

class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        openCt = 0
        closeCt = 0
        copied = list(s)
        for i,c in enumerate(s):
            if c=="(":
                openCt+=1
            else:
                closeCt+=1
            if openCt==closeCt:
                copied[i-(openCt*2-1)]=""
                copied[i]=""
                openCt=0
                closeCt=0
        return "".join(copied)
